packages/lodestar-cli/src/cmds/account/cmds/wallet/create.ts

```diff
@@ -0,0 +1,105 @@
+import fs from "fs";
+import path from "path";
+import {CommandBuilder} from "yargs";
+import * as bip39 from "bip39";
+import {randomPassword, writeFile600Perm,YargsError} from "../../../../util";
+import {WalletManager} from "../../../../wallet";
+import {getAccountPaths} from "../../paths";
+import {IAccountWalletOptions} from "./options";
+
+export const command = "create";
+
+export const description = "Creates a new HD (hierarchical-deterministic) EIP-2386 wallet";
+
+interface IWalletCreateOptions extends IAccountWalletOptions {
+  name: string;
+  passphraseFile: string;
+  type: string;
+  mnemonicOutputPath?: string;
+}
+
+export const builder: CommandBuilder<{}, IWalletCreateOptions> = {
+  name: {
+    description: "The wallet will be created with this name. It is not allowed to \
+create two wallets with the same name for the same --base-dir.",
+    alias: ["n"],
+    demandOption: true,
+    type: "string"
+  },
+
+  passphraseFile: {
+    description: "A path to a file containing the password which will unlock the wallet. \
+If the file does not exist, a random password will be generated and saved at that \
+path. To avoid confusion, if the file does not already exist it must include a \
+'.pass' suffix.",
+    alias: ["passphrase-file", "p"],
+    demandOption: true,
+    type: "string"
+  },
+
+  type: {
+    description: `The type of wallet to create. Only HD (hierarchical-deterministic)
+wallets are supported presently.`,
+    choices: ["hd"],
+    default: "hd",
+    type: "string"
+  },
+
+  mnemonicOutputPath: {
+    description: "If present, the mnemonic will be saved to this file",
+    alias: ["mnemonic-output-path"],
+    type: "string"
+  }
+};
+
+export async function handler(options: IWalletCreateOptions): Promise<void> {
+  const name = options.name;
+  const type = options.type;
+  const passphraseFile = options.passphraseFile;
+  const mnemonicOutputPath = options.mnemonicOutputPath;
+  const accountPaths = getAccountPaths(options);
+
+  // Create a new random mnemonic.
+  const mnemonic = bip39.generateMnemonic();
+
+  // Create a random password if the file does not exist.
+  if (!fs.existsSync(passphraseFile)) {
+    if (path.parse(passphraseFile).ext !== ".pass") {
+      throw new YargsError("passphraseFile must end with .pass, make sure to not provide the actual password");
+    }
+    writeFile600Perm(passphraseFile, randomPassword());
+  }
+  const password = fs.readFileSync(passphraseFile, "utf8");
+
+  const walletManager = new WalletManager(accountPaths);
+  const wallet = walletManager.createWallet(name, type, mnemonic, password);
+
+  if (mnemonicOutputPath) {
+    writeFile600Perm(mnemonicOutputPath, mnemonic);
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(`
+Your wallet's 12-word BIP-39 mnemonic is:
+
+\t${mnemonic}
```

Comment:
I wonder if we should print this since it would stay in bash history

Reply:
How could we provide the mnemonic to the user otherwise?

Reply:
It's inside wallet store, but we can leave this for now